R/io-parsers.R

```diff
@@ -135,6 +135,15 @@ parse_repertoire <- function(.filename, .mode, .nuc.seq, .aa.seq, .count,
     }
   }
 
+  # if V, D or J columns are missing in the data, add empty columns
+  for (genes_var in c(".vgenes", ".dgenes", ".jgenes")) {
+    if (is.na(get(genes_var))) {
```

Comment:
In this line if checks if first element of genes_var is NA or not; it is not cool because it makes unnecessary all other part of vector;
If you want to check is such columns are in the df, you can use new_cols_vector %in% colnames(df) construction;